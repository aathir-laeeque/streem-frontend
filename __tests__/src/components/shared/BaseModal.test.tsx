import { BaseModal } from '#components';
import React from 'react';
import { create, ReactTestRenderer } from 'react-test-renderer';

describe('<BaseModal />', () => {
  let snapshot: ReactTestRenderer;

  const props = {
    closeAllModals: jest.fn(),
    closeModal: jest.fn(),
    onSecondary: jest.fn(),
    title: 'Base Modal Test',
    primaryText: 'Primary',
    secondaryText: 'Cancel',
    onPrimary: jest.fn(),
  };

  beforeAll(() => {
    const component = (
      <BaseModal {...props}>
        <div />
      </BaseModal>
    );
    snapshot = create(component);
  });

  it('should render and match the snapshot', () => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
