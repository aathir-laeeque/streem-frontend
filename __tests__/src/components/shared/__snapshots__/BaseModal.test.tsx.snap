// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseModal /> should render and match the snapshot 1`] = `
<div
  className="sc-fznKkj eYPgoK"
>
  <div
    className="openup"
    id="modal-container"
  >
    <div
      className="modal-background"
    >
      <div
        className="modal"
      >
        <div
          className="modal-header"
        >
          <h2>
            Base Modal Test
          </h2>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "fontSize": 20,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          className="modal-body"
        >
          <div />
        </div>
        <div
          className="modal-footer"
        >
          <div
            className="modal-footer-buttons"
          >
            <button
              className="sc-fzokOt izNgjE"
              onClick={[Function]}
              style={
                Object {
                  "fontWeight": 600,
                  "padding": "5px 16px",
                }
              }
            >
              Cancel
            </button>
            <button
              className="sc-fznZeY fxACvd"
              onClick={[Function]}
              style={
                Object {
                  "fontWeight": 600,
                  "marginRight": 0,
                }
              }
            >
              Primary
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
